### ISSUE TITLE
W3D4T2 train function doesn't use test data, affecting bonus exercises

### ISSUE THREAD

**Issue by romanpogodin** [NeuromatchAcademy/course-content#500]
In tutorial 2, the `train` function doesn't use test inputs at all, and uses train labels instead of test in the regularized case. It only affects the bonus part of the tutorial.

Original implementation, with problems commented:
```
# Track progress
if (i+1) % (n_iter // 10) == 0 or i==0:
    if test_data is not None and test_labels is not None:
        #
        # y_pred is never re-computed
        # 
        if L2_penalty>0 or L1_penalty>0:
            loss = custom_loss(y_pred, 
                               train_labels,  # must be test
                               weights, L2_penalty, L1_penalty)
        else:
            loss = custom_loss(y_pred, test_labels)
        test_loss[i] = loss.item()
        print(f'iteration {i+1}/{n_iter} | train loss: {train_loss[i]:.4f} | test loss: {test_loss[i]:.4f}')
    else:
    print(f'iteration {i+1}/{n_iter} | train loss: {train_loss[i]:.4f}')
```

Fix:
```   
# Track progress
if (i+1) % (n_iter // 10) == 0 or i==0:
    if test_data is not None and test_labels is not None:
        y_pred = net(test_data)
        if L2_penalty>0 or L1_penalty>0:
            loss = custom_loss(y_pred, test_labels, weights, L2_penalty, L1_penalty)
        else:
            loss = custom_loss(y_pred, test_labels)
        test_loss[i] = loss.item()
        print(f'iteration {i+1}/{n_iter} | train loss: {train_loss[i]:.4f} | test loss: {test_loss[i]:.4f}')
    else:
        print(f'iteration {i+1}/{n_iter} | train loss: {train_loss[i]:.4f}')
```

**Comment by ebatty** [comment]
Fixed, thanks!
